compiler/rustc_parse/src/parser/item.rs

```diff
@@ -227,7 +227,7 @@ impl<'a> Parser<'a> {
             // STATIC ITEM
             self.bump(); // `static`
             let m = self.parse_mutability();
-            let (ident, ty, expr) = self.parse_item_global(Some(m))?;
+            let (ident, _, ty, expr) = self.parse_item_global(Some(m))?;
```

Comment:
In the current implementation they can actually be non-empty for static items since we're now doing parse error recovery for generic statics (respecting `.may_recover()`). I chose to do that spontaneously as a drive-by diagnostic improvement. See [parser/generic-statics.stderr](https://github.com/rust-lang/rust/blob/39d85200a9910c318eb56b34ab3e02e7f5d5eb86/tests/ui/parser/generic-statics.stderr) for the result. Shouldn't be too costly.